plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1016,6 +1020,10 @@ private static Optional<EncryptionMaterialsProvider> createEncryptionMaterialsPr
 
     private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configuration conf)
     {
+        if (anonymousRequestsEnabled) {
```

Comment:
I'm not sure if we really need a new config property. Can we reuse `trino.s3.credentials-provider` with `org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider`?

Reply:
Wouldn't that only work if all tables in the catalog were stored in public buckets?

But your suggestion gave me an idea:

We can have a custom CredentialProvider which chains the default credential provider chain and adds `org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider` to the end and use it as the default value.

Reply:
@ebyhr `org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider` wouldn't work, since the class specified by `trino.s3.credentials-provider` needs to:

> provide a two-argument constructor that takes a `java.net.URI` and a Hadoop `org.apache.hadoop.conf.Configuration` as arguments

(from the [Trino Hive/S3 docs](https://trino.io/docs/current/connector/hive-s3.html#custom-s3-credentials-provider)).

However, I just discovered a configuration change that _does_ work, with no code change required:

```
hive.s3.signer-class=com.amazonaws.auth.NoOpSigner
```

This simply replaces the default S3 signer with one that does nothing - exactly what we need for anonymous access to public buckets!

Note that you also need the following in `conf/metastore-site.xml`:

```xml
<property>
    <name>fs.s3a.aws.credentials.provider</name>
    <value>org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider</value>
</property>
```

I'll open a ticket and file a PR over at https://github.com/trinodb/docs.trino.io

@hashhar The problem is, `TrinoS3FileSystem` doesn't know if the configured credentials will work until it later tries to use them. The way the connector is currently written, all buckets accessed by the connector have to be accessible with the same set of credentials.